CommonTools/UtilAlgos/interface/ProductFromFwdPtrProducer.h

```diff
@@ -25,7 +25,7 @@ namespace edm {
 
 
   template < class T, class H = ProductFromFwdPtrFactory<T> >
-  class ProductFromFwdPtrProducer : public edm::EDProducer {
+    class ProductFromFwdPtrProducer : public edm::global::EDProducer<> {
```

Comment:
All that exist so far don't use global variables.

Since a new factory method is made/destructed each iteration of the for-loop, unless a static variable is used somewhere in the factory (which I think our static-analyzer would notice) there's not too much evil to be done in the first place.